### ISSUE TITLE
xmr-stak-2.10.4 AMD GPU mining does not start

### ISSUE THREAD

**Issue by entak** [fireice-uk/xmr-stak#2401]
**Most execution issues are caused by driver problems. Please use the [xmr-stak sub-reddit](https://www.reddit.com/r/XmrStak/) to ask for help instead of opening an issue here.**

**Comment by lordhugo7880** [comment]
I would first try clearing your OpenCl cache. This entails deleting the contents of %userprofile%\AppData\Local\.openclcache\

I would also recommend updating your AMD driver to a version that's part of the Adrenaline series. The latest driver currently is version 19.4.1, but for example 18.5.2 also works good. Another thing you can try is to install the AMD APP SDK 3.0, which if you use it you need to reinstall after installing an update for the AMD driver. (This is a workaround, because you shouldn't need the SDK for OpenCL to function.)

**Comment by lordhugo7880** [comment]
I would first try clearing your OpenCl cache. This entails deleting the contents of %userprofile%\AppData\Local.openclcache\

I would also recommend updating your AMD driver to a version that's part of the Adrenalin series. The latest driver currently is version 19.4.1, but for example 18.5.2 also works good. Another thing you can try is to install the AMD APP SDK 3.0, which if you use it you need to reinstall after installing an update for the AMD driver. (This is a workaround, because you shouldn't need the SDK for OpenCL to function.)

**Comment by lordhugo7880** [comment]
I would first try clearing your OpenCl cache. This entails deleting the contents of %userprofile%\AppData\Local\.openclcache\

I would also recommend updating your AMD driver to a version that's part of the Adrenalin series. The latest driver currently is version 19.4.1, but for example 18.5.2 also works good. Another thing you can try is to install the AMD APP SDK 3.0, which if you use it you need to reinstall after installing an update for the AMD driver. (This is a workaround, because you shouldn't need the SDK for OpenCL to function.)

**Comment by lordhugo7880** [comment]
I would first try clearing your OpenCl cache. This entails deleting the contents of %userprofile%\AppData\Local\.openclcache\

I would also recommend updating your AMD driver to a version that's part of the Adrenalin series. The latest driver currently is version 19.4.1, but for example 18.5.2 also works good. Another thing you can try is to install the AMD APP SDK 3.0, which if you use it you need to reinstall after installing an update for the AMD driver. (This is a workaround, because you shouldn't need the SDK for OpenCL to function.)

**Comment by lordhugo7880** [comment]
Local \ .openclcache \

**Comment by Spudz76** [comment]
If you have the Intel IGD drivers installed, their OpenCL can get in the way and crash things

Use the default windows included 2D-only driver for the IGD (it doesn't do OpenCL) or search and destroy the Intel OpenCL Client Driver DLL (I forget the file name right now).  Or uninstall all display drivers with DDU and then install AMD only (let windows default driver handle the Intel, assuming you're using it as the display?)  If you aren't using the IGD at all then disable it/don't install a driver.

**Comment by Spudz76** [comment]
Does it generate an amd.txt when one doesn't exist?
What does it say?

**Comment by lordhugo7880** [comment]
Good point about the Intel OpenCL. Concerning the Intel OpenCL, it's also possible to disable that from the Windows registry by going to the keys described on this page:

https://stackoverflow.com/questions/35230731/how-to-disable-registered-opencl-platforms-on-windows

and then giving the DWORD or DWORDs with the path to Intel OpenCL a value of 1 (usually in the registry 1 is on and 0 is off, but here it's the opposite)

**Comment by lordhugo7880** [comment]
To also use the AMD OpenCl for mining on the CPU, it may be necessary to use the older OpenCL dll provided by the AMD APP SDK 3.0.

**Comment by lordhugo7880** [comment]
To also use the AMD OpenCL for mining on the CPU, it may be necessary to use the older OpenCL dll provided by the AMD APP SDK 3.0.

**Comment by Spudz76** [comment]
Thanks for that link I searched for it a while hoping to be complete with my answer, but gave up.

I have SFP shut off along with all the other space and time wasting protections so I just went and deleted the client DLL that the driver installed, so even if it was mapped in the registry it was forced-off-forever via not being there at all.  Tho I bet SFP or other new Win10 annoyances (oops, assistances) would probably restore it for your own protection.

Either method allows for using the full Intel drivers (for better Open**G**L and D3D or whatever) but without their broken OpenCL client DLL in the mix.

Does old AMD OpenCL with all-CPU support actually perform better than raw CPU miners?

I think I tried the Intel CPU OpenCL and it of course doesn't work just like their GPU OpenCL never works.  Then I may not have bothered trying the AMD-OpenCL on CPU assuming non-GPU would not work just like the Intel stuff didn't.  And also using OpenCL on nVidia is generally always slower than "native" CUDA, so I figured transitively the raw CPU backends would be fastest versus some "extra" layer in the mix.  But OpenCL-CPU probably knows more about how to handle alloc and layout and dynamic compilation for performance and might actually gain over the less agile CPU backend.

**Comment by Spudz76** [comment]
Also maybe dummy plug, if the AMD card has no screen connected to it, they used to not work without also having an output (even if the output goes nowhere, ala dummy plug).  Because they won't enable a GPU with no graphical output (even for computation).

I don't know if that is even a thing anymore on RX series.  I always ran Linux for AMD mining since its driver does not care about screens connected / more specifically you can tell it a fake screen and force it on (ignoring physical detection and EDID missing).

**Comment by lordhugo7880** [comment]
I don't know if AMD OpenCL on CPU performs better than Intel OpenCL on CPU or vice versa, but the topic starter talked about also mining on the CPU, which if it's an Intel CPU of course wouldn't work if the Intel OpenCL got disabled. Plus I've read that at some point AMD removed all CPU support from their OpenCL implementation but did consider to possibly add it back again in the future. So therefore my suggestion to use the older AMD OpenCL dll.
